torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -950,6 +968,32 @@ def call_function(
             should_flatten_outputs=True,
         )
 
+        subgraph_example_value = [
+            proxy.node.meta["example_value"] for proxy in body_r.as_proxy()
+        ]
+
+        def _compute_stride(subgraph_out: torch.Tensor) -> Tuple[int]:
+            if subgraph_out.ndim == 0:
+                return (1,)
+            else:
+                return (
+                    subgraph_out.size()[0] * subgraph_out.stride()[0],
+                    *subgraph_out.stride(),
+                )
+
+        with tx.output.fake_mode:
+            map_example_out = [
+                torch.empty_strided(
+                    size=(sample_shape[0], *t.size()),
+                    stride=_compute_stride(t),
+                    dtype=t.dtype,
+                    layout=t.layout,
+                    device=t.device,
+                    requires_grad=t.requires_grad,
+                )
+                for t in subgraph_example_value
+            ]
```

Comment:
If the semantics of map are "for-loop then stack", then I would expect the output tensor to be contiguous instead of having this stride. Also, can we add a test that the strides of map under FakeTensor are the same as the strides from eager?

Reply:
yeah, stack does make the output contiguous. Added a test to compare eager run map_dense results has the same metadata as the `example_values`.